dataland-email-service/src/test/kotlin/org/dataland/datalandemailservice/email/EmailSenderTest.kt

```diff
@@ -1,66 +1,80 @@
-import org.junit.jupiter.api.assertThrows
-import org.mockito.ArgumentMatchers.any
 import org.mockito.Mockito.mock
 import org.mockito.Mockito.`when`
+import org.mockito.kotlin.any
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
 
 class EmailSenderTest {
-    private class EmailSendException : RuntimeException()
-
     private val senderContact = EmailContact("sender@example.com")
-    private val receiversContact = EmailContact("receiver@example.com")
-    private val mockMailjetClient = mock(MailjetClient::class.java)
-    private val mockMailjetRequest = mock(MailjetRequest::class.java)
+    private val emailContent = EmailContent("", "", "")
+
+    private lateinit var mockMailjetClient: MailjetClient
+    private lateinit var emailSender: EmailSender
+    private lateinit var emailSubscriptionTracker: EmailSubscriptionTracker
 
-    private fun callEmailSender(): EmailSender {
-        `when`(mockMailjetClient.post(any() ?: mockMailjetRequest)).thenThrow(EmailSendException())
-        val emailSender = EmailSender(mockMailjetClient)
-        return emailSender
+    private val unsubscribedEmailAddress = "abc@123.com"
+    private val missingEmailAddress = "xyz@123.com"
+    private val subscribedEmailAddress = "def@123.com"
+
+    @BeforeEach
+    fun setup() {
+        mockMailjetClient = mock(MailjetClient::class.java)
+        `when`(mockMailjetClient.post(any())).thenReturn(MailjetResponse(200, "{}"))
+        emailSubscriptionTracker = mock(EmailSubscriptionTracker::class.java)
+        `when`(emailSubscriptionTracker.emailIsSubscribed(unsubscribedEmailAddress)).thenReturn(false)
+        `when`(emailSubscriptionTracker.emailIsSubscribed(missingEmailAddress)).thenReturn(null)
+        `when`(emailSubscriptionTracker.emailIsSubscribed(subscribedEmailAddress)).thenReturn(true)
+        emailSender = EmailSender(mockMailjetClient, emailSubscriptionTracker)
     }
 
     @Test
-    fun `check if the mail sender works as expected`() {
-        val senderContact = EmailContact("sender@dataland.com")
+    fun `check if the mail is send to subscribed email address`() {
+        val receiver = EmailContact(subscribedEmailAddress)
         val email =
             Email(
-                senderContact, listOf(senderContact),
-                listOf(), EmailContent("", "", ""),
+                senderContact, listOf(receiver),
+                listOf(), emailContent,
             )
-        val emailSender = callEmailSender()
-        assertThrows<EmailSendException> {
-            emailSender.sendEmailWithoutTestReceivers(email)
-        }
+        emailSender.filterReceiversAndSentEmail(email)
+        verify(mockMailjetClient, times(1)).post(any())
     }
 
     @Test
-    fun `check if the suppressing of test receivers works as expected  `() {
-        val senderCC = EmailContact("CC@somethingelse.comn")
+    fun `check if the mail is send to email address not stored in the database`() {
```

Comment:
```suggestion
    fun `check if the email is sent to email address not stored in the database`() {
```

Also, but I guess this relates to returning true in the "isEmailsubscribed" function: Why do you send an email to a person not in the SubscriptionRepo?  Do I understand correctly that the subscriptionID is only required when sending ClaimOwnerShip Mails; for other types of mails, the subscriptionID is not necessary and thus, recipients also don't need to have one?